sites/geohub/src/components/LayerList.svelte

```diff
@@ -31,90 +34,91 @@
         return
       }
       const styleInfo = await res.json()
-      restoredStyle = styleInfo.style
+
       if (styleInfo.layers) {
-        $layerList = styleInfo.layers
+        let legendState = {}
+        //console.log('restoring ', styleInfo.layers.length)
         const style: StyleSpecification = styleInfo.style
+
+        styleInfo.layers.map((el) => {
+          const layerStyle = style.layers.find((l) => l.id === el.id)
+          const cmap = layerStyle?.['colormap']
+          const classification = layerStyle?.['classification']
+          const lid = layerStyle?.['id']
+          if (cmap && classification && lid) {
+            //reuse state
+            legendState[lid] = { classification: classification, colorMapName: cmap }
+          }
+
+          //console.log(JSON.stringify(legendState))
+        })
+
+        // the state has been restored this assignement will trigger UI rendering
+
         $map.setStyle(style)
         $map.flyTo({
           center: [style.center[0], style.center[1]],
           zoom: style.zoom,
           bearing: style.bearing,
           pitch: style.pitch,
         })
+        $layerList = styleInfo.layers
         activeTab = TabNames.LAYERS
+        return legendState
       } else {
         $page.url.searchParams.delete('style')
-        goto(`?${$page.url.searchParams.toString()}`)
+        //goto(`?${$page.url.searchParams.toString()}`)
```

Comment:
please uncomment this code. This code is going to remove `style` query param from URL if it is invalid saved style.